test/SampleAuthorization.js

```diff
@@ -977,31 +977,31 @@ describe("2250: Sample Authorization", () => {
       .post("/api/v3/Samples/" + sampleId1 + "/attachments")
       .send(TestData.AttachmentCorrect)
       .set("Accept", "application/json")
-      .expect(TestData.CreationForbiddenStatusCode);
+      .expect(TestData.UnauthorizedStatusCode);
   });
 
   it("0420: adds an attachment as an unauthenticated user to sample 2, which should fail", async () => {
     return request(appUrl)
       .post("/api/v3/Samples/" + sampleId2 + "/attachments")
       .send(TestData.AttachmentCorrect)
       .set("Accept", "application/json")
-      .expect(TestData.CreationForbiddenStatusCode);
+      .expect(TestData.UnauthorizedStatusCode);
   });
 
   it("0430: adds an attachment as an unauthenticated user to sample 3, which should fail", async () => {
     return request(appUrl)
       .post("/api/v3/Samples/" + sampleId3 + "/attachments")
       .send(TestData.AttachmentCorrect)
       .set("Accept", "application/json")
-      .expect(TestData.CreationForbiddenStatusCode);
+      .expect(TestData.UnauthorizedStatusCode);
   });
 
   it("0440: adds an attachment as an unauthenticated user to sample 4, which should fail", async () => {
     return request(appUrl)
       .post("/api/v3/Samples/" + sampleId4 + "/attachments")
       .send(TestData.AttachmentCorrect)
       .set("Accept", "application/json")
-      .expect(TestData.CreationForbiddenStatusCode);
+      .expect(TestData.UnauthorizedStatusCode);
```

Comment:
Why are we changing from CreationForbiddenStatusCode to UnauthorizedStatusCode?

Reply:
To give you a brief context, we now have two policy guards for APIs, PoliciesGuard and AuthenticatedPoliciesGuard.

PoliciesGuard is used for endpoints that unauthenticated users can access. Permission is checked at the instance scope, and errors are controlled by the permissionChecker function, which throws a Forbidden 403 error if access is denied.

AuthenticatedPoliciesGuard is used for endpoints that only authenticated users can access. Errors are controlled by the guard level + instance level, and a Unauthorized 401 error is thrown when an unauthenticated user sends a request to an endpoint decorated with AuthenticatedPoliciesGuard.

So, for endpoints that can be accessed by unauthenticated users, we should use Forbidden 403. For endpoints that require authentication, we should use Unauthorized 401.

Reply:
OK

Reply:
POST method should only be called by authenticated users & with right permission. In this case, it will return 401 Unauthorized